CMake/RealmCore.cmake

```diff
@@ -390,8 +392,16 @@ macro(build_realm_sync)
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION_RELEASE ${sync_server_library_release})
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION ${sync_server_library_release})
 
-    find_package(PkgConfig)
-    pkg_check_modules(YAML QUIET yaml-cpp)
+    if(REALM_PLATFORM STREQUAL "Android")
+      ExternalProject_Add(YAML
+        GIT_REPOSITORY "git@github.com:jbeder/yaml-cpp.git"
+        CMAKE_ARGS "-DCMAKE_SYSTEM_NAME=Android -DCMAKE_ANDROID_ARCH_ABI=${ANDROID_ABI} -DCMAKE_ANDROID_NDK=${ANDROID_NDK}"
```

Comment:
If you do it this way (and not using a toolchain) then you have to bump the minimum required version to 3.7.
I think a toolchain would still be the way to go at this point.